src/profile/webid.ts

```diff
@@ -91,21 +93,20 @@ export async function getProfileAll<
   T extends SolidDataset & WithServerResourceInfo
 >(
   webId: WebId,
-  options: Partial<
+  options?: Partial<
     typeof internal_defaultFetchOptions & {
       webIdProfile: T;
     }
-  > = internal_defaultFetchOptions
+  >
 ): Promise<ProfileAll<T>> {
-  const {
-    fetch,
-    webIdProfile = (await getSolidDataset(webId, { fetch })) as T,
-  } = options;
-
+  const authFetch = options?.fetch ?? defaultFetch;
+  const webIdProfile =
+    options?.webIdProfile ??
+    ((await getSolidDataset(webId, { fetch: unauthenticatedFetch })) as T);
```

Comment:
Aside, it might be nice to change `getSolidDataset` to accept `T` as a generic, such that we could just do:
```suggestion
    (await getSolidDataset<T>(webId, { fetch: unauthenticatedFetch }));
```

Reply:
Actually, I'd argue that this is an issue with the function signature rather than `getSolidDataset`: in the case when `getSolidDataset` is needed to fetch the WebID, the generic type is unbound, and the return type should be `SolidDataset & WithServerResourceInfo`, which is the return type of `getSolidDataset` in the first place. I'll see if this can be cleaned up with a little overload.

Reply:
That was easy enough: https://github.com/inrupt/solid-client-js/pull/1530